sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -416,7 +426,7 @@ class RelationalGroupedDataset protected[sql](
         new RelationalGroupedDataset(
           df,
           groupingExprs,
-          RelationalGroupedDataset.PivotType(pivotColumn.expr, values.map(Literal.apply)))
+          RelationalGroupedDataset.PivotType(pivotColumn.expr, values.map(lit(_).expr)))
```

Comment:
What do you think about `map(lit).map(_.expr)` instead?

Reply:
Don't see any advantages of this. It is longer and slower.